src/commands/deploy.ts

```diff
@@ -214,39 +212,82 @@ async function verifyRuntimeIsCompatible(localRuntime: ProjectRuntime, ui: IAzur
     }
 }
 
-async function tryPublishCSharpProject(deployFsPath: string, outputChannel: vscode.OutputChannel, telemetryProperties: TelemetryProperties): Promise<void> {
+async function runPreDeployTask(deployFsPath: string, telemetryProperties: TelemetryProperties, language: ProjectLanguage, isZipDeploy: boolean): Promise<void> {
+    let taskName: string | undefined = getFuncExtensionSetting(preDeployTaskSetting, deployFsPath);
+    if (!isZipDeploy) {
+        // We don't run pre deploy tasks for non-zipdeploy since that stuff should be handled by kudu
+
+        if (taskName) {
+            // We only need to warn if they have the setting defined
+            ext.outputChannel.appendLine(localize('ignoringPreDeployTask', 'WARNING: Ignoring preDeployTask "{0}" for non-zip deploy.', taskName));
+        }
+
+        return;
+    }
+
+    let isTaskNameDefined: boolean = true;
+    if (!taskName) {
+        isTaskNameDefined = false;
+        switch (language) {
+            case ProjectLanguage.CSharp:
+                taskName = publishTaskId;
+                break;
+            case ProjectLanguage.JavaScript:
+                taskName = installExtensionsId;
+                break;
+            default:
+                return; // preDeployTask not needed
```

Comment:
Need telemetry for this?

Reply:
Don't think so. Also it can be inferred from our existing telemetry (we already have the language and we're tracking the 'preDeployTask' if it's defined in settings).